core_impl/src/main/java/io/opencensus/implcore/stats/MutableViewData.java

```diff
@@ -491,12 +496,50 @@ public MutableAggregation apply(Distribution arg) {
   }
 
   private static final class CreateSumData implements Function<MutableSum, AggregationData> {
+
+    private final Measure measure;
+
+    private CreateSumData(Measure measure) {
+      this.measure = measure;
+    }
+
     @Override
-    public AggregationData apply(MutableSum arg) {
-      return SumData.create(arg.getSum());
+    public AggregationData apply(final MutableSum arg) {
+      return measure.match(
+          new CreateSumDataDouble(arg),
+          new CreateSumDataLong(arg),
```

Comment:
Suggestion: These function objects ignore their input, so I think you could write this as:

```java
      return measure.match(
          Functions.<AggregationData>returnConstant(SumDataDouble.create(arg.getSum())),
          Functions.<AggregationData>returnConstant(SumDataLong.create(Math.round(arg.getSum()))),
          Functions.<AggregationData>throwAssertionError());
```

Reply:
Nice suggestion, updated